src/http/mod.rs

```diff
@@ -0,0 +1,89 @@
+//! HTTP transport modules.
+
+#[cfg(feature = "simple_http")]
+pub mod simple_http;
+
+#[cfg(feature = "minreq_http")]
+pub mod minreq_http;
+
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+use std::{fmt, io};
+
+/// The default TCP port to use for connections.
+/// Set to 8332, the default RPC port for bitcoind.
+pub const DEFAULT_PORT: u16 = 8332;
+
+/// The Default SOCKS5 Port to use for proxy connection.
+/// Set to 9050, the default RPC port for tor.
+// Currently only used by `simple_http` module, here for consistency.
+#[cfg(feature = "proxy")]
+pub const DEFAULT_PROXY_PORT: u16 = 9050;
+
+/// The different authentication methods for the client.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub enum Auth {
+    /// No authentication.
+    None,
+    /// Basic authentication using (user, password).
+    UserPass(String, String),
+    /// Basic authentication using cookie file (contains 'user:password').
+    CookieFile(PathBuf),
```

Comment:
This is really bitcoin-core specific stuff. "Basic Auth" just has a `user:pass` format and the cookiefile thing is just Bitcoind that we do in bitcoincore-rpc. I don't think it belongs here. A string field "basic_auth_header" or something should be sufficient.

Reply:
I wouldn't mind having a constructor that takes a `PathBuf` and reads a `user:pass` format out of it to construct a `UserPass`. This would let people easily use Bitcoin Core without needing the full force of bitcoincore-rpc.

But I agree we shouldn't treat the cookie file stuff as a first-class thing or directly have any data structures for it.

Reply:
This suggestion has been implemented in the PR
```rust
/// Reads a 'user:password' string from `cookie`.
///
/// This function exists to support reading the cookie file used by `bitcoind`.
pub fn read_auth_from_file(cookie: &Path) -> Result<(String, String), CookieError> {
    ...
```